### PR TITLE
feat(dlp): added a sample for de-identify replace 

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description
- Added a sample for [de-identify replace](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-replace) with unit test case.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid/deid_replace.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		Parent:           fmt.Sprintf("projects/%s/locations/global", projectID),
+		DeidentifyConfig: deIdentifyConfig,
+		InspectConfig:    inspectConfig,
+		Item:             item,
+	}
+
+	// Send the request.
+	resp, err := client.DeidentifyContent(ctx, req)
+	if err != nil {
+		return fmt.Errorf("DeidentifyContent: %v", err)
```
Comment (telpirion) [inline dlp/snippets/deid/deid_replace.go:105]: nit: use `%w` formatting or don't wrap error.

File: dlp/snippets/deid/deid_test.go [hunk 2]
```diff
@@ -136,6 +136,23 @@ func TestDeidentifyExceptionList(t *testing.T) {
...
 }
 
+func TestDeIdentifyWithReplacement(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	input := "My name is Alicia Abernathy, and my email address is aabernathy@example.com."
+	infoType := []string{"EMAIL_ADDRESS"}
+	replaceVal := "[email-address]"
+	want := "output : My name is Alicia Abernathy, and my email address is [email-address]."
+
+	buf := new(bytes.Buffer)
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:146]: issue: don't use `new`. Instead write `var buf bytes.Buffer`.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid/deid_test.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Resolved.

File: dlp/snippets/deid/deid_replace.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_replace.go]: Resolved.

**Review by annadata — APPROVED** [review]
LGTM

**Pull request merged**